libbeat/cmd/instance/metrics.go

```diff
@@ -106,36 +106,81 @@ func reportBeatCPU(_ monitoring.Mode, V monitoring.Visitor) {
 	V.OnRegistryStart()
 	defer V.OnRegistryFinished()
 
-	totalCPUUsage, err := getCPUUsage()
+	totalCPUUsage, totalCPUUserTicks, totalCPUSystemTicks, totalCPUTicks, err := getCPUUsage()
 	if err != nil {
 		logp.Err("Error retrieving CPU percentages: %v", err)
 		return
 	}
 
+	userTime, systemTime, err := process.GetOwnResourceUsageTime()
+	if err != nil {
+		logp.Err("Error retrieving CPU usage time: %v", err)
+		return
+	}
+
+	monitoring.ReportNamespace(V, "user", func() {
+		monitoring.ReportInt(V, "ticks", int64(totalCPUUserTicks))
+		monitoring.ReportInt(V, "time", userTime)
+	})
+	monitoring.ReportNamespace(V, "system", func() {
+		monitoring.ReportInt(V, "ticks", int64(totalCPUSystemTicks))
+		monitoring.ReportInt(V, "time", systemTime)
+	})
 	monitoring.ReportNamespace(V, "total", func() {
 		monitoring.ReportFloat(V, "value", totalCPUUsage)
+		monitoring.ReportInt(V, "ticks", int64(totalCPUTicks))
+		monitoring.ReportInt(V, "time", userTime+systemTime)
```

Comment:
I see how `monitoring.ReportX` can become very cumbersome to use. Using go-structform, we could create a `structform.Visitor` mapping to `monitoring.Visitor`. Then we could write:

```
type allCPUStats struct {
  User cpuStats
  System cpuStats
  Total cpuStats
}

type cpuStats struct {
  Time int64
}

...

monitoring.ReportStruct(V, allCPUStat{
  User: cpuStat{ ... }
  System: cpuStat{ ... }
  Total: cpuStat{ ... }
})
```

WDYT?